doc-content/enterprise-only/openshift/customimage-smartrouter-proc.adoc

```diff
@@ -0,0 +1,139 @@
+[id='customimage-smartrouter-proc_{context}']
+:componentimage: smartrouter
+:component: Smart Router
+= Creating a custom Smart Router image with an additional JAR file to implement custom routing
+
+By default, Smart Router routes requests based on the container alias. If severak {KIE_SERVERS} provide a servcie with teh same container alias, Smart Router balances the load between them.
+
+In some cases, custom routing functionality is required. You can create a custom class to implement the custom routing and then create a custom Smart Router image with the class. You can push this image into your custom registry and then use it to deploy Smart Router.
+
+.Prerequisites
+
+* A JDK and Apache Maven are installed.
+* The project for deploying {PRODUCT} is created in your {OPENSHIFT} environment
+* You know the route for the {OPENSHIFT} image registry and have the permission to push images into the registry. For instructions about configuring the registry, see https://access.redhat.com/documentation/en-us/openshift_container_platform/4.5/html/registry/index[_Registry_] in {OPENSHIFT} product documentation.
+
+.Procedure
+
+. Download the `{PRODUCT_FILE}-container-sources.zip` product deliverable file from the {PRODUCT_DOWNLOAD_LINK}[Software Downloads] page of the Red Hat Customer Portal.
+. Extract the contents of the file and change to the `/src/rhpam-7-openshift-image/quickstarts/router-ext` directory.
+. Modify the sample source of the router extension in this directory as necessary. The existing code implements simple routing based on the version of the container.
+. Build the source code with Maven:
++
+[subs="attributes,verbatim,macros,quotes"]
+----
+mvn clean package
+----
++
+The build process generates the following JAR file: `target/router-ext-0.0.1-SNAPSHOT.jar` 
++
+. Create a work directory for creating the custom image, copy the generated JAR file into the directory, and then change to the directory, for example:
++
+[subs="attributes,verbatim,macros,quotes"]
+----
+mkdir /tmp/smartrouter
+cp target/router-ext-0.0.1-SNAPSHOT.jar /tmp/smartrouter
+cd /tmp/smartrouter
+----
++
+include::customimage-download-step.adoc[]
+. To extract the `openshift-launch.sh` file from the official Smart Router image, enter the following command:
```

Comment:
MAybe add more context here. like

To customize the Smart ROuter image to use the custom routing .... you'll need to extract the 'openshift-launch.sh' file from the official smart router image..